cmd/kubeadm/app/master/clusterroles.go

```diff
@@ -0,0 +1,47 @@
+package master
+
+import (
+	"fmt"
+
+	api "k8s.io/kubernetes/pkg/api/v1"
+	rbac "k8s.io/kubernetes/pkg/apis/rbac/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+)
+
+// CreateBootstrapRBACClusterRole creates the necessary ClusterRole for bootstrapping
+func CreateBootstrapRBACClusterRole(clientset *clientset.Clientset) error {
+	clusterRole := rbac.ClusterRole{
+		// a role to use for setting up a proxy
+		ObjectMeta: api.ObjectMeta{Name: "system:kubelet-bootstrap"},
+		Rules: []rbac.PolicyRule{
+			rbac.NewRule("get").Groups("").Resources("nodes").RuleOrDie(),
+			rbac.NewRule("create", "watch").Groups("certificates.k8s.io").Resources("certificatesigningrequests").RuleOrDie(),
```

Comment:
> @liggitt The purpose why we're doing this is not to be yet another deployment type, it's to be the lean and configurable core. 

setting up users and permissions that do not exist in all deployments makes kubeadm a type of deployment, albeit a lightweight one.

> I'm fine with kubeadm: if there absolutely is no other way, but do we have any alternatives?

This is just a convention to make sure you don't conflict with any future roles defined in https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/

As roles are proved out here, it is likely they can migrate to system roles that do get autoloaded, though kubeadm supporting deploying older versions of kubernetes might make it tricky to depend on more than the most fundamental roles.